src/codeu/chat/DerbyStore.java

```diff
@@ -103,15 +147,17 @@ public boolean checkUsernameExists(String name) throws SQLException {
 		return user == null;
 	}
 	
-	public String userLoginCheck(String name, String password) throws SQLException {
+	@Override
+	public User userLogin(String name, String password) throws SQLException, IOException {
 		PreparedStatement checkValidUserTest = conn.prepareStatement(checkValidUser);
 		checkValidUserTest.setString(1, name);
 		checkValidUserTest.setString(2, password);
 		ResultSet user = checkValidUserTest.executeQuery();
 		
-		return (user != null) ? user.getString(1) : null;
+		return (user != null) ? new User(Uuid.parse(user.getString(1)), user.getString(2), Time.fromMs(user.getLong(3))) : null;
 	}
 	
+	@Override
 	public void addUser(User u) throws SQLException {
 		PreparedStatement addUser = conn.prepareStatement(addUserInfo);
 		addUser.setString(1, removeCharsInUuid(u.id.toString()));
```

Comment:
What if the User has the same username or ID as an existing User? Is that check handled elsewhere? Even if it is, a badguy could still submit bad stuff against your server, so it's a good idea to have here as well.